### PR TITLE
Merge for authentication routes

### LINKED ISSUE #44: Password recovery
Create laravel controllers for password recovery

Content:
- Create route for password recovery page
- Define and implement password recovery workflow
- Test workflow against unit tests

As a user, I should be able to reset my account's password so that I can log into the service again.

depends on: #57 
**Estimation: 1h**

### PR DESCRIPTION
**Pull request by Serphentas**
This PR for #38 #42 #44

### CODE REVIEW

File: src/app/Http/Controllers/Auth/LoginController.php
Comment (kayoumido) [inline src/app/Http/Controllers/Auth/LoginController.php]: This is an API, so you don't have the need for an index page.

File: src/routes/web.php
Comment (kayoumido) [inline src/routes/web.php]: This change breaks Vue.js routing system... Please but it back.

File: src/app/Http/Controllers/Auth/RegisterController.php
Comment (kayoumido) [inline src/app/Http/Controllers/Auth/RegisterController.php]: Your should return a `UserResource`

It should look something like this 
```php
return new UserResource(User::create([..]));
```
Resources aren't yet in the project. Once #94 is merged, you can rebase your branch so you can use them

File: src/routes/web.php
Comment (kayoumido) [inline src/routes/web.php]: Just a guess, but this should be in `src/routes/api.php`. You should check with @fabio10000

**Review by kayoumido — CHANGES_REQUESTED** [review]
We're building an SPA with Laravel being the API

File: src/app/Http/Controllers/Auth/LoginController.php
Comment (fabio10000) [inline src/app/Http/Controllers/Auth/LoginController.php]: this protection should appear in the `routes/api.php` file instead of here
you should see the laravel documentation about routing and middlewares to a better understanding: https://laravel.com/docs/6.x/routing#route-group-middleware
```php
// example of what you should do
Route::middleware('auth:api')->group(function() {
    // routes for authenticated actions
});
```

File: src/app/Http/Controllers/Auth/LoginController.php
Comment (fabio10000) [inline src/app/Http/Controllers/Auth/LoginController.php]: Laravel should never return a view as it is our API system (except for the only entry point of the vue application), those methods should always return a resource object : https://laravel.com/docs/6.x/eloquent-resources#generating-resources

File: src/app/Http/Controllers/Auth/RegisterController.php
Comment (fabio10000) [inline src/app/Http/Controllers/Auth/RegisterController.php]: Same here never return a view from laravel

File: src/routes/web.php
Comment (fabio10000) [reply src/routes/web.php]: you should not use that helper to generate routes as it generates a non API routing and we don't want that as Laravel is an API application.
You should look at the routes created by Auth:routes with `php artisan routes:list` and report the usefull routes to the file `routes/api.php`
Also watchout the default Auth controller are not meant to be used within an API application you have to adapt some several points

File: src/routes/web.php
Comment (fabio10000) [inline src/routes/web.php]: this on is the only route that we should have in our `routes/web.php`

**Review by fabio10000 — CHANGES_REQUESTED** [review]
It's disapointing that you don't hever had a look at the laravel documentation who explains all those things.
you should have a look at this link before further modifications and don't hesitate to ask me if there are some points that you don't understand rather than doing things like that

File: src/app/Http/Controllers/Auth/LoginController.php
Comment (Serphentas) [reply src/app/Http/Controllers/Auth/LoginController.php]: Fixed in 82175c3

File: src/routes/web.php
Comment (Serphentas) [reply src/routes/web.php]: Fixed in 299808a

File: src/app/Http/Controllers/Auth/LoginController.php
Comment (Serphentas) [reply src/app/Http/Controllers/Auth/LoginController.php]: Fixed in c37e406

File: src/app/Http/Controllers/Auth/LoginController.php
Comment (Serphentas) [reply src/app/Http/Controllers/Auth/LoginController.php]: See comments above

File: src/app/Http/Controllers/Auth/RegisterController.php
Comment (Serphentas) [reply src/app/Http/Controllers/Auth/RegisterController.php]: Fixed in 5bcb65f

File: src/routes/web.php
Comment (Serphentas) [reply src/routes/web.php]: See comments above

File: src/app/Http/Controllers/Auth/RegisterController.php
Comment (Serphentas) [reply src/app/Http/Controllers/Auth/RegisterController.php]: Fixed in 645b86445a62a2f926c36d73fefab764ee1b7ffd

File: src/routes/web.php
Comment (Serphentas) [reply src/routes/web.php]: Fixed in 3212365

File: src/app/Http/Controllers/Auth/RegisterController.php
Comment (fabio10000) [inline src/app/Http/Controllers/Auth/RegisterController.php:73]: data should be validated before used to create an user

File: src/routes/api.php
Comment (fabio10000) [inline src/routes/api.php:18]: this route is a frontend route that should not appear here

File: src/routes/api.php
Comment (fabio10000) [inline src/routes/api.php:23]: this route is a frontend route that should not appear here

File: src/routes/api.php
Comment (fabio10000) [inline src/routes/api.php:26]: this route is a frontend route that should not appear here

File: src/routes/api.php
Comment (fabio10000) [inline src/routes/api.php:28]: this route is a frontend route that should not appear here

File: src/routes/api.php
Comment (fabio10000) [inline src/routes/api.php:31]: this route is a frontend route that should not appear here

File: src/app/Http/Controllers/Auth/RegisterController.php
Comment (fabio10000) [reply src/app/Http/Controllers/Auth/RegisterController.php:73]: using the function validator at line 52 for example

**Review by fabio10000 — CHANGES_REQUESTED** [review]

**Comment by kayoumido** [comment]
done in #114

**Pull request closed without merging**
